sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -211,7 +211,10 @@ private void processNonCachedEvent(final @NotNull SentryBaseEvent event) {
   private void setThreads(final @NotNull SentryEvent event) {
     if (event.getThreads() != null) {
       for (SentryThread thread : event.getThreads()) {
-        thread.setCurrent(MainThreadChecker.isMainThread(thread));
+        Boolean isCurrentThread = thread.isCurrent();
+        if (isCurrentThread == null || !isCurrentThread) {
+          thread.setCurrent(MainThreadChecker.isMainThread(thread));
```

Comment:
The null check is already sufficient. In the `!isCurrentThread` case it would still override previous data and thus, it wouldn't fix the linked bug.
Example:
isCurrent is false, then the code in the check would be executed and the isCurrent is then set to true (assuming Sentry runs on the main thread). That would be wrong for the given bug report, since the thread data comes out of the Dart VM which has it's own threads

Reply:
Thanks @ueman I've updated the code.